controllers/serving/inference_controller.go

```diff
@@ -273,6 +294,80 @@ func (ir *InferenceReconciler) syncTrafficDistribution(inf *servingv1alpha1.Infe
 	return nil
 }
 
+// syncPredictorTrafficDistribution syncs traffic-routing resources to ensure traffic is distributed by the
+// user configuration required.
+// For example, user A serves two models and distribute 90% traffic to model A.1 and the left to
+// model A.2, the network topology looks like this:
+//
+// User
+//  |---request---> VirtualService
+//                        |--- 90% ---> Deploy-Of-Model-A.1
+//                        |--- 10% ---> Deploy-Of-Model-B.1
+func (ir *InferenceReconciler) syncPredictorTrafficDistribution(inf *servingv1alpha1.Inference, index int, predictor *servingv1alpha1.PredictorSpec) error {
+	igInCluster := networkingv1beta1.Ingress{}
+	igExists := true
+	igName := genPredictorName(inf, predictor)
+	err := ir.client.Get(context.Background(), types.NamespacedName{Namespace: inf.Namespace, Name: igName}, &igInCluster)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	if err != nil && errors.IsNotFound(err) {
+		igExists = false
+	}
+
+	// At this point, ingress is not found and create a new one.
+
+	ratios := computePredictorTrafficRatios(inf)
+	// Traffic will firstly hit istio-less ingress gateway and then be dispatched to different
+	// destinated model servers.
+	ig := networkingv1beta1.Ingress{
```

Comment:
are we creating an ingress for each predicator ?
or should it be one ingress for all predicators ? @SimonCqk

Reply:
Usually l think canary requires multiple ingress entries, I prefer to create one ingress for each predicator )